fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/StringExpNodeVisitor.java

```diff
@@ -0,0 +1,406 @@
+        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String div(String left, String right) {
+        StringBuilder result = new StringBuilder();
+        result.append(left);
+        result.append(" / ");
+        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String isNull(String expr) {
+        StringBuilder result = new StringBuilder();
+        result.append(expr);
+        result.append(" IS NULL");
+        return result.toString();
+    }
+
+    @Override
+    public String isNotNull(String expr) {
+        StringBuilder result = new StringBuilder();
+        result.append(expr);
+        result.append(" IS NOT NULL");
+        return result.toString();
+    }
+
+    @Override
+    public String between(String leftValue, String rightValue) {
+        StringBuilder result = new StringBuilder();
+        result.append(" BETWEEN ");
+        result.append(leftValue);
+        result.append(" AND ");
+        result.append(rightValue);
+        return result.toString();
+    }
+
+    @Override
+    public String column(String tableAlias, String columnName) {
+        StringBuilder result = new StringBuilder();
+        if (tableAlias != null) {
+            result.append(tableAlias);
+            result.append(".");
+        }
+        result.append(columnName);
+        return result.toString();
+    }
+
+    @Override
+    public String bindMarker(Double value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new DoubleBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(Long value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new LongBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(String value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new StringBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(Instant value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new InstantBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(Integer value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new IntegerBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String in(String left, List<String> args) {
+        StringBuilder result = new StringBuilder();
+        result.append(left);
+        result.append(" IN (");
+        result.append(String.join(",", args));
+        result.append(")");
+        return result.toString();
+    }
+
+    @Override
+    public String like(String left, String right) {
+        StringBuilder result = new StringBuilder();
+        result.append(left);
+        result.append(" LIKE ");
+        result.append(right);
```

Comment:
Are these expressions before the `like`? I'd assume the right should be encoded

Reply:
the left and right could be a complex expression. We only need to encode literals at the point they are rendered.